testing/demo/pkg/setup/tendermint.go

```diff
@@ -0,0 +1,114 @@
+package main
+
+import (
+	"context"
+	"fmt"
+
+	"os"
+	"os/exec"
+
+	"github.com/celestiaorg/celestia-zkevm-ibc-demo/testing/demo/pkg/utils"
+	"github.com/cosmos/solidity-ibc-eureka/abigen/ics26router"
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/ethclient"
+)
+
+// CreateTendermintLightClient creates the Tendermint light client on the EVM roll-up.
+func CreateTendermintLightClient() error {
+	err := utils.CheckSimappNodeHealth(simappRPC, 10)
+	if err != nil {
+		return fmt.Errorf("simapp node is not healthy, please ensure it is running correctly: %w", err)
+	}
+
+	err = utils.CheckEthereumNodeHealth(ethereumRPC)
+	if err != nil {
+		return fmt.Errorf("ethereum node is not healthy, please ensure it is running correctly: %w", err)
+	}
+
+	if err := deployEurekaContracts(); err != nil {
+		return err
+	}
+
+	addresses, err := utils.ExtractDeployedContractAddresses()
+	if err != nil {
+		return err
+	}
+
+	if err := addClientToRouter(addresses); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// deployEurekaContracts deploys all of the IBC Eureka contracts (including the
+// SP1 ICS07 Tendermint light client contract) on the EVM roll-up.
+func deployEurekaContracts() error {
+	cmd := exec.Command("forge", "script", "E2ETestDeploy.s.sol:E2ETestDeploy", "--rpc-url", "http://localhost:8545", "--private-key", "0x82bfcfadbf1712f6550d8d2c00a39f05b33ec78939d0167be2a737d691f33a6a", "--broadcast")
+	cmd.Env = append(cmd.Env, "PRIVATE_KEY=0x82bfcfadbf1712f6550d8d2c00a39f05b33ec78939d0167be2a737d691f33a6a")
+	cmd.Dir = "./solidity-ibc-eureka/scripts"
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+
+	fmt.Printf("Deploying IBC Eureka smart contracts on the EVM roll-up...\n")
+	if err := cmd.Run(); err != nil {
+		return fmt.Errorf("failed to deploy contracts: %v", err)
+	}
+	fmt.Printf("Deployed IBC Eureka smart contracts on the EVM roll-up.\n")
+
+	addresses, err := utils.ExtractDeployedContractAddresses()
+	if err != nil {
+		return err
+	}
+	fmt.Printf("Contract Addresses: \n%v\n", addresses)
+
+	return nil
+}
+
+// addClientToRouter adds the Tendermint light client to the router contract on
+// the EVM roll-up.
+//
+// Note this also registers the counterparty.
+func addClientToRouter(addresses utils.ContractAddresses) error {
```

Comment:
Didn't realise there needed to be a transaction to link the two smart contracts

Reply:
Yep. Adds it here: https://github.com/rootulp/celestia-zkevm-ibc-demo/blob/80d92ce1393a523173aac577220f5b38563aa26f/testing/demo/pkg/setup/tendermint.go#L97